### PR TITLE
[SPARK-30231][SQL][PYTHON][FOLLOWUP] Make error messages clear in PySpark df.explain

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

This pr is a followup of #26861 to address minor comments from @viirya.

### Why are the changes needed?

For better error messages.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Manually tested.

### CODE REVIEW

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -305,11 +305,11 @@ def explain(self, extended=None, mode=None):
...
-            argtypes = [
-                str(type(arg)) for arg in [extended, mode] if arg is not None]
-            raise TypeError(
-                "extended (optional) and mode (optional) should be a bool and str; "
-                "however, got [%s]." % ", ".join(argtypes))
+            if extended is not None:
+                errMsg = "extended should be provided as bool, got {0}".format(type(extended))
+            else:  # For mode case
+                errMsg = "mode should be provided as str, got {0}".format(type(mode))
+            raise TypeError(errMsg)
```
Comment (maropu) [inline python/pyspark/sql/dataframe.py]: How about this error handling? @viirya https://github.com/apache/spark/pull/26861#discussion_r357726827

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 2]
```diff
@@ -547,6 +547,7 @@ class Dataset[T] private[sql](
     }
   }
 
+  // This method intends to be called from PySpark DataFrame
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: https://github.com/apache/spark/pull/26861#discussion_r357727679

**Comment by maropu** [comment]
also cc: @HyukjinKwon

File: python/pyspark/sql/dataframe.py (hunk 1 above)
Comment (viirya) [reply python/pyspark/sql/dataframe.py]: either `extended (optional) should be ...` or `extended, if given, should be ...`. same for `mode`.

**Review by viirya — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Oh, I didn't know you opened a PR already. Sure, this way works too.

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/dataframe.py (hunk 1 above)
Comment (maropu) [reply python/pyspark/sql/dataframe.py]: Ah, ok. I'll update. (but, I'm away from my pc, so do tonight.)

Comment (maropu) [reply python/pyspark/sql/dataframe.py]: Does the latest commit satisfy you comment? The current exception behaviours are as follows;
```
>>> spark.range(1).explain(extended=1)
TypeError: extended (optional) should be provided as bool, got <class 'int'>

>>> spark.range(1).explain(mode=1)
TypeError: mode (optional) should be provided as str, got <class 'int'>

>>> spark.range(1).explain(extended=1, mode=1)
Exception: extended and mode can not be specified simultaneously
```

Comment (viirya) [reply python/pyspark/sql/dataframe.py]: LGTM.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.
Thank you, @maropu , @viirya , @HyukjinKwon !

**Comment by maropu** [comment]
Thank for taking care, @dongjoon-hyun !

**Pull request closed without merging**